### ISSUE TITLE
Signed integer overflow in atomic perf tests

### ISSUE THREAD

**Issue by dalg24** [kokkos/kokkos#5563]
Clang's UndefinedBehaviorSanitizer is raising 2 warnings in `KokkosCore_PerformanceTest_Atomic`
```
<kokkos>/core/perf_test/test_atomic.cpp:320:13: runtime error: signed integer overflow: 2147450880 + 65536 cannot be represented in type 'int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior <kokkos>/core/perf_test/test_atomic.cpp:320:13 in
<kokkos>/core/perf_test/test_atomic.cpp:361:12: runtime error: signed integer overflow: 2147450880 + 65536 cannot be represented in type 'int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior <kokkos>/core/perf_test/test_atomic.cpp:361:12 in

**Comment by cz4rs** [comment]
@ajpowelsnl let me know if you need help with this. Also the line numbers in the description are out of date since this file has seen some changes in the meantime :wink:

**Comment by cz4rs** [comment]
@ajpowelsnl Can you please check for some fresh occurrences and post the warnings here? (I don't have access to nightlies)

**Comment by dalg24** [comment]
Should get resolved by #7104
